colossalai/auto_parallel/solver/strategy/matmul_strategy_generator.py

```diff
@@ -0,0 +1,312 @@
+        input_size = self._compute_size_in_bytes(strategy, "input")
+        other_size = self._compute_size_in_bytes(strategy, "input")
+
+        if "bias" in self.operation_data_mapping:
+            bias_size = self._compute_size_in_bytes(strategy, "bias")
+        else:
+            bias_size = 0
+        output_size = self._compute_size_in_bytes(strategy, "output")
+
+        fwd_mem_cost = MemoryCost(activation=output_size, parameter=other_size + bias_size)
+        bwd_mem_cost = MemoryCost(activation=input_size + other_size + bias_size, parameter=other_size)
+        total_mem_cost = MemoryCost(activation=input_size + 2 * output_size + bias_size,
+                                    parameter=other_size + bias_size)
+        memory_cost = TrainCycleItem(fwd=fwd_mem_cost, bwd=bwd_mem_cost, total=total_mem_cost)
+        strategy.memory_cost = memory_cost
+
+    def generate(self) -> List[ShardingStrategy_V2]:
+        strategies = []
+
+        # SS = SR x RS
+        strategies.append(self.split_lhs_space_rhs_space(0, 1))
+        strategies.append(self.split_lhs_space_rhs_space(1, 0))
+
+        # SR = SS x SR
+        strategies.append(self.split_lhs_space_both_contract(0, 1))
+        strategies.append(self.split_lhs_space_both_contract(1, 0))
+
+        # RS = RS x SS
+        strategies.append(self.split_rhs_space_both_contract(0, 1))
+        strategies.append(self.split_rhs_space_both_contract(1, 0))
+
+        # RR= RS x SR
+        strategies.append(self.recompute_split_both_contract(0))
+        strategies.append(self.recompute_split_both_contract(1))
+
+        # RS = RR x RS
+        strategies.append(self.split_rhs_space_only(0))
+        strategies.append(self.split_rhs_space_only(1))
+
+        # S01R = S01R x RR
+        strategies.append(self.split_lhs_1st_dim_1d(0, 1))
+
+        # RR = RS01 x S01R
+        strategies.append(self.split_lhs_2nd_dim_1d(0, 1))
+
+        # RS01 = RR x RS01
+        strategies.append(self.split_rhs_2nd_dim_1d(0, 1))
+
+        # update mete info on cost
+        for strategy in strategies:
+            self.update_communication_cost(strategy)
+            self.update_compute_cost(strategy)
+            self.update_memory_cost(strategy)
+
+        return strategies
+
+    def split_lhs_space_rhs_space(self, mesh_dim_0, mesh_dim_1):
+        # handle case SS = SR x RS
+        name = f'S{mesh_dim_0}S{mesh_dim_1} = S{mesh_dim_0}R x RS{mesh_dim_1}'
+        dim_partition_dict = {
+            self.operation_data_mapping['input']: {
+                0: [mesh_dim_0]
+            },
+            self.operation_data_mapping['other']: {
+                self.dim_q: [mesh_dim_1]
+            },
+            self.operation_data_mapping['bias']: {
+                -1: [mesh_dim_1]
+            },
+            self.operation_data_mapping['output']: {
+                0: [mesh_dim_0],
+                -1: [mesh_dim_1]
+            },
+        }
+        communication_actions = {}
```

Comment:
For training mission, all reduce communication is required during computing the gradient of parameters, and gradient of activation.

Reply:
done.